scene/animation/tween.cpp

```diff
@@ -75,12 +75,17 @@ Ref<PropertyTweener> Tween::tween_property(Object *p_target, NodePath p_property
 	ERR_FAIL_COND_V_MSG(!valid, nullptr, "Tween invalid. Either finished or created outside scene tree.");
 	ERR_FAIL_COND_V_MSG(started, nullptr, "Can't append to a Tween that has started. Use stop() first.");
 
-#ifdef DEBUG_ENABLED
-	Variant::Type property_type = p_target->get_indexed(p_property.get_as_property_path().get_subnames()).get_type();
-	ERR_FAIL_COND_V_MSG(property_type != p_to.get_type(), Ref<PropertyTweener>(), "Type mismatch between property and final value: " + Variant::get_type_name(property_type) + " and " + Variant::get_type_name(p_to.get_type()));
-#endif
+	bool access_ok;
+	Variant::Type target_type = p_target->get_indexed(p_property.get_as_property_path().get_subnames(), &access_ok).get_type();
+	ERR_FAIL_COND_V_MSG(!access_ok, nullptr, "Can't access property " + p_property);
```

Comment:
The message could be reworded and include more details, like "Can't find property 'property' in object 'object'.